### DISCUSSION TITLE
Tycho 2.7.1 bugfix release

### DISCUSSION THREAD

**Discussion by akurtakov** [eclipse-tycho/tycho#819] [Announcements]
We are preparing a bugfix release of Tycho 2.7.1 to include some important fixes. 

For details of included bugfixes, see the [release notes](https://github.com/eclipse/tycho/blob/tycho-2.7.x/RELEASE_NOTES.md#271), please let us know if you are aware of other regressions you would like to have included.

Also please help by testing the SNAPSHOTS build. To use it, change your Tycho version to 2.7.1-SNAPSHOT and add the following snippet to your pom.

```
<pluginRepositories>
    <pluginRepository>
      <id>tycho-snapshots</id>
      <url>https://repo.eclipse.org/content/repositories/tycho-snapshots/</url>
    </pluginRepository>
</pluginRepositories>
```

You can find the current planned release date and progress [here](https://github.com/eclipse/tycho/milestone/6). 

**We like to thank [Compart AG ](https://www.compart.com/) who kindly sponsoring this bugfix release and some of the included fixes.**

Regards,
Tycho team

**Comment by merks** [comment]
Perhaps it would be good if the 2.7.1 release used Eclipse 4.23 so that it can be used to build products that have PGP-signed content and could be used for EPP's 2022-06 builds.  Right now 2.7.1 fails as describe here because of a p2 bug in 4.22:

https://github.com/eclipse/tycho/discussions/651#discussioncomment-2477776

**Reply by laeubi** [reply]
@merks is there a simple case to reproduce the issue? I think it would be useful to have an integration test for this.

**Comment by phermsdorf** [comment]
Works for me. 
(the maven repository still get's messed up for pre 2.7 tycho builds but that's probably a known issue)

Thanks for your work!

**Comment by merks** [comment]
**Any** product build that includes a PGP signed bundle will reproduce this problem.  Oomph's installer product is feature based so I added this to the installer's main feature:
```
   <plugin
         id="org.mockito.mockito-core"
         download-size="0"
         install-size="0"
         version="0.0.0"
         unpack="false"/>
```
The TP then must include either Eclipse 4.23 or 4.24.  What I was also trying to test was that the final artifacts.xml in the installed product included these keys:
```
	public static final String PGP_SIGNER_KEYS_PROPERTY_NAME = "pgp.publicKeys"; //$NON-NLS-1$
	public static final String PGP_SIGNATURES_PROPERTY_NAME = "pgp.signatures"; //$NON-NLS-1$
```
That's because without these two keys for each PGP-signed bundle because otherwise the bundle will be displayed as unsigned content.  So you also have to configure this:
```
          <execution>
            <id>materialize-products</id>
            <goals>
              <goal>materialize-products</goal>
            </goals>
            <configuration>
...
              <source>repository</source>
            </configuration>
          </execution>
```

**Reply by laeubi** [reply]
Hi Ed,

I created [a testcase here](https://github.com/eclipse/tycho/pull/826) that

1. creates a product with the `org.mockito.mockito-core` included
2. check that the generated repository contains the pgp keys.

The test succeeds with Tycho 2.7.0 and Tycho 3.0.0-SNAPSHOT without a problem.

**Reply by merks** [reply]
I reviewed it but now I'm doubtful it actually materializes a product, which is the part that fails with an NPE.  In any case, the test does not check artifacts.xml of the product but rather that of the repository and it doesn't check the pgp.publicKeys property which is what is needed in the about dialog...

**Comment by HannesWell** [comment]
Tycho 2.7.1-SNAPSHOT works well for m2e: https://github.com/eclipse-m2e/m2e-core/pull/626
At least the part besides the build-extension (which still can't be tested as SNAPSHOT can it?).
I also tested this with my employers build and this works as well too.

**Reply by laeubi** [reply]
You need the snapshot repo in your settings.xml see https://github.com/eclipse/tycho/wiki#configure-snapshots-repository-globally

**Reply by HannesWell** [reply]
Thanks for that hint. I adjusted the snapshot build and replayed it with the modified JenkinsFile. 
Still works well. 👍🏽

**Comment by laeubi** [comment]
Tycho 2.7.1 has been released and is available from Maven Central repository.

🆕 https://github.com/eclipse/tycho/blob/tycho-2.7.1/RELEASE_NOTES.md
🏷️ https://github.com/eclipse/tycho/tree/tycho-2.7.1
👔 https://projects.eclipse.org/projects/technology.tycho/releases/2.7.1
🙏 contributors who contributed patches for this release:

- 	Alexander Kurtakov
- 	Christoph Läubrich
- 	dependabot[bot]
- 	Mickael Istria
- 	Sebastian Ratz

and thanks and to everyone who helped us with testing the staged version and especially Compart AG who kindly sponsoring this bugfix release and some of the included fixes.

Regards,
The Tycho Team

**Comment by laeubi** [comment]
There was one issue identified after the release see https://github.com/eclipse/tycho/issues/876 so I now created a new 2.7.2-SNAPSHOT for anyone affected by this.
